src/rpc/chain_api.rs

```diff
@@ -377,3 +442,187 @@ fn load_api_messages_from_tipset(
 
     Ok(messages)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use PathChange::{Apply, Revert};
+
+    use crate::{
+        blocks::{chain4u, Chain4U, RawBlockHeader},
+        db::{car::PlainCar, MemoryDB},
+        networks::{self, ChainConfig},
+    };
+
+    #[test]
+    fn revert_to_ancestor_linear() {
+        let store = ChainStore::calibnet();
+        chain4u! {
+            in store.blockstore();
+            [_genesis = store.genesis_block_header()]
+            -> [a] -> [b] -> [c, d] -> [e]
+        };
+
+        // simple
+        assert_path_change(&store, b, a, [Revert(&[b])]);
+
+        // from multi-member tipset
+        assert_path_change(&store, [c, d], a, [Revert(&[c, d][..]), Revert(&[b])]);
+
+        // to multi-member tipset
+        assert_path_change(&store, e, [c, d], [Revert(e)]);
+
+        // over multi-member tipset
+        assert_path_change(&store, e, b, [Revert(&[e][..]), Revert(&[c, d])]);
+
+        // TODO(aatifsyed): how should we handle incomplete `TipsetKey`s from the user?
+        // assert_path_change(&store, e, c, [Revert(e)]); // fails
+        // assert_path_change(&store, c, a, [Revert(&[c]), Revert(&[b])]); // passes
```

Comment:
What do we do with this? Should we replicate Lotus' behaviour (if it makes sense)?

Reply:
Tested this now